include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -539,6 +540,50 @@ class open_addressing_impl {
     return counter.load_to_host(stream);
   }
 
+  // TODO docs
+  template <typename Container, typename Predicate>
+  void rehash(Container const& container, Predicate const& is_filled, cuda_stream_ref stream)
+  {
+    this->rehash_async(container, is_filled, stream);
+    stream.synchronize();
+  }
+
+  template <typename Container, typename Predicate>
+  void rehash(extent_type extent,
+              Container const& container,
+              Predicate const& is_filled,
+              cuda_stream_ref stream)
+  {
+    this->rehash_async(extent, container, is_filled, stream);
+    stream.synchronize();
+  }
+
+  template <typename Container, typename Predicate>
+  void rehash_async(Container const& container, Predicate const& is_filled, cuda_stream_ref stream)
+  {
+    this->rehash_async(this->storage_.window_extent(), container, is_filled, stream);
+  }
+
+  template <typename Container, typename Predicate>
+  void rehash_async(extent_type extent,
+                    Container const& container,
+                    Predicate const& is_filled,
+                    cuda_stream_ref stream)
+  {
+    auto const old_storage = std::move(this->storage_);
+    new (&storage_) storage_type{extent, this->allocator()};
+    this->clear_async(stream);
+
+    auto const num_windows = old_storage.num_windows();
+    if (num_windows == 0) { return; }
+    auto constexpr block_size = cuco::detail::default_block_size();
+    auto constexpr stride     = cuco::detail::default_stride();
+    auto const grid_size      = cuco::detail::int_div_ceil(num_windows, stride * block_size);
```

Comment:
```suggestion
    auto const grid_size      = cuco::detail::grid_size(num_windows);
```
https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/utility/cuda.hpp#L40-L46

Reply:
Addressed in [d29b8a9](https://github.com/NVIDIA/cuCollections/pull/380/commits/d29b8a948ce63b3f696c3279d2440e1f922df2e4) . Although I don like using the default arguments, since we use the `block_size` and `stride` again when configuring the kernel. This is more readable I think.

Reply:
Fair concern. I won't enforce this then.

The goal of having those helper functions is to avoid duplicates but then duplicates enhance readability...